install.sh

```diff
@@ -4,48 +4,69 @@
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 install_vim() {
-    # Clone Vundle first
-    git clone https://github.com/VundleVim/Vundle.vim.git \
-        ${HOME}/.vim/bundle/Vundle.vim
+    vundle_path="${HOME}/.vim/bundle/Vundle.vim"
+
+    if [ ! -d "$vundle_path" ]; then
+      # Clone Vundle first
+      git clone https://github.com/VundleVim/Vundle.vim.git $vundle_path
+    else
+      cd $vundle_path
+      git fetch origin
+      git pull
+    fi
 
     # Link .vimrc
-    ln -s ${DIR}/vim/.vimrc ${HOME}
+    ln -sf ${DIR}/vim/.vimrc ${HOME}
 
     # Link Russian-jcukenmac keymap
     mkdir -p ${HOME}/.vim/keymap
-    ln -s ${DIR}/vim/.vim/keymap/russian-jcukenmac.vim \
+    ln -sf ${DIR}/vim/.vim/keymap/russian-jcukenmac.vim \
         ${HOME}/.vim/keymap/
 
     # Link additional tabular commands
     mkdir -p ${HOME}/.vim/bundle/tabular/after/plugin/
-    ln -s ${DIR}/vim/.vim/bundle/tabular/after/plugin/my_tabular_commands.vim \
+    ln -sf ${DIR}/vim/.vim/bundle/tabular/after/plugin/my_tabular_commands.vim \
         ${HOME}/.vim/bundle/tabular/after/plugin/
 
     # Link filetype definitions
-    ln -s ${DIR}/vim/.vim/ftdetect ${HOME}/.vim/
+    ln -sf ${DIR}/vim/.vim/ftdetect ${HOME}/.vim/
 }
 
 install_tmux() {
-    # Clone Tmux Plugin Manager
-    git clone https://github.com/tmux-plugins/tpm ${HOME}/.tmux/plugins/tpm
+    tmux_plugin_path="${HOME}/.tmux/plugins/tpm"
+
+    if [ ! -d "$tmux_plugin_path" ]; then
+      # Clone Tmux Plugin Manager
+      git clone https://github.com/tmux-plugins/tpm $tmux_plugin_path
+    else
+      cd $tmux_plugin_path
+      git fetch origin
+      git pull
+    fi
 
     # Link .tmux.conf
-    ln -s ${DIR}/tmux/.tmux.conf ${HOME}
+    ln -sf ${DIR}/tmux/.tmux.conf ${HOME}
 }
 
 # Bash
-ln -s ${DIR}/bash/.bashrc ${HOME}
```

Comment:
@valAV Why do you remove this command?

`.bashrc` is executed when an interactive non-login shell starts.

Reply:
macOS terminal by default runs a login-shell and _.bashrc_ will never be used.